core/src/credential.rs

```diff
@@ -105,19 +186,29 @@ impl Credential {
     }
 }
 
-#[derive(Debug, serde::Serialize, serde::Deserialize, Clone)]
-pub struct VerifiablePresentation {
-    #[serde(flatten)]
-    pub presentation: Presentation,
-    pub proof: crate::proof::DataIntegrityProof,
+impl FromStr for Presentation {
+    type Err = super::error::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let p = Presentation::from_json_str(s)?;
+        Ok(p)
+    }
 }
 
-#[derive(Debug, serde::Serialize, serde::Deserialize, Clone)]
-pub struct Presentation {
-    #[serde(rename = "@context")]
-    pub context: VerificationContext,
-    #[serde(rename = "verifiableCredential")]
-    pub verifiable_credential: Vec<VerifiableCredential>,
+impl FromStr for VerifiablePresentation {
+    type Err = super::error::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let vp = VerifiablePresentation::from_json_str(s)?;
+        Ok(vp)
+    }
+}
+
+impl std::fmt::Display for VerifiablePresentation {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let vp = self.to_json_string().unwrap();
```

Comment:
same thing here, should be be crashing if this unwrap fails?

Reply:
See comment here - https://github.com/knox-networks/ssi/pull/167#discussion_r1406883667